src/Compilers/Core/Portable/SourceGeneration/IncrementalWrapper.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Wraps an incremental generator in a dummy <see cref="ISourceGenerator"/> interface.
+    /// </summary>
+    /// <remarks>
+    /// Used to unify loading incremental generators with older ISourceGenerator style ones. There are various places
+    /// that assume there is a 1 to 1 mapping between generator instance and type. While it was never an explicit guarantee
+    /// enough downstream consumers take a dependency on this behavior that is worth preserving. This wrapper allows us
+    /// to continue to maintain that mapping while still wrapping the type.
+    /// </remarks>
+    /// <typeparam name="TIncrementalGenerator">The type of the incrmental generator being wrapped</typeparam>
+    internal sealed class IncrementalToSourceGeneratorWrapper<TIncrementalGenerator> : IncrementalGeneratorWrapper, ISourceGenerator
```

Comment:
```suggestion
    internal sealed class IncrementalGeneratorWrapper<TIncrementalGenerator> : IncrementalGeneratorWrapper, ISourceGenerator
```

So the non-generic and generic names are the same? Or that might just be more confusing.